cmd/common/instances.go

```diff
@@ -0,0 +1,206 @@
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cols := columns.MustCreateColumns[GadgetInfo]()
+			cols.MustAddColumn(columns.Attributes{
+				Name:         "ID",
+				Visible:      true,
+				EllipsisType: ellipsis.End,
+				Order:        10,
+				Width:        12,
+				MaxWidth:     12,
+				MinWidth:     12,
+			}, func(g *GadgetInfo) any {
+				if g.pg == nil {
+					return ""
+				}
+				return g.pg.Id[:12]
+			})
+			cols.MustAddColumn(columns.Attributes{
+				Name:         "Name",
+				Visible:      true,
+				EllipsisType: ellipsis.End,
+				Order:        20,
+			}, func(g *GadgetInfo) any {
+				if g.pg == nil {
+					return ""
+				}
+				return g.pg.Name
+			})
+			cols.MustAddColumn(columns.Attributes{
+				Name:         "Tags",
+				Visible:      true,
+				EllipsisType: ellipsis.End,
+				Order:        30,
+			}, func(g *GadgetInfo) any {
+				if g.pg == nil {
+					return ""
+				}
+				return strings.Join(g.pg.Tags, ",")
+			})
+			cols.MustAddColumn(columns.Attributes{
+				Name:         "Gadget",
+				Visible:      true,
+				EllipsisType: ellipsis.End,
+				Order:        50,
+			}, func(g *GadgetInfo) any {
+				if g.pg == nil {
+					return ""
+				}
+				return g.pg.GadgetInfo.ImageName
+			})
+			// cols.MustAddColumn(columns.Attributes{
+			// 	Name:         "Nodes",
+			// 	Visible:      true,
+			// 	EllipsisType: ellipsis.End,
+			// 	Order:        60,
+			// }, func(g *GadgetInfo) any {
+			// 	return strings.Join(g.pg.GadgetInfo.Nodes, ",")
+			// })
+
+			formatter := textcolumns.NewFormatter(cols.GetColumnMap())
+			fmt.Println(formatter.FormatHeader())
+
+			gadgets, err := runtime.GetGadgetInstances(context.Background(), runtimeParams)
+			if err != nil {
+				return err
+			}
+			for _, gadget := range gadgets {
+				gi := &GadgetInfo{pg: gadget}
+				fmt.Println(formatter.FormatEntry(gi))
+				if showDetails {
+					for k, v := range gi.pg.GadgetInfo.ParamValues {
+						if len(v) > 128 {
+							v = v[:128] + "…"
+						}
+						fmt.Printf("  %-32s %s\n", k, v)
+					}
+				}
+			}
+			return nil
+		},
+	}
+	listCmd.PersistentFlags().BoolVarP(&showDetails, "details", "", false, "show details")
+	AddFlags(listCmd, runtimeParams, nil, runtime)
+	rootCmd.AddCommand(listCmd)
+
+	deleteCmd := &cobra.Command{
+		Use:          "delete",
+		Aliases:      []string{"d"},
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return fmt.Errorf("missing id")
+			}
+
+			gadgetInstanceID := args[0]
+			gadgetInstance, err := findGadgetInstance(gadgetInstanceID)
+			if err != nil {
+				return fmt.Errorf("finding gadget: %w", err)
+			}
+
+			return runtime.RemoveGadgetInstance(context.Background(), runtimeParams, gadgetInstance.Id)
+		},
+	}
+	AddFlags(deleteCmd, runtimeParams, nil, runtime)
+	rootCmd.AddCommand(deleteCmd)
+
+	stopCmd := &cobra.Command{
+		Use:          "stop",
+		Aliases:      []string{"s"},
```

Comment:
Similar to above question should I be able to attach to stopped gadget? e.g 

```
$  sudo ./gadgetctl run profile_blockio --detachable
$ sudo ./gadgetctl list
ID           NAME                                                                    TAGS                                                                    GADGET                                                                 
0e3b203821d6 elegant_matsumoto                                                                                                                               trace_open                                                             
17ee372cbb9f affectionate_bardeen                                                                                                                            profile_blockio  
$ sudo ./gadgetctl attach 17ee372cbb9f
latency
        µs               : count    distribution
         0 -> 1          : 0        |                                        |
         1 -> 2          : 0        |                                        |
         2 -> 4          : 0        |                                        |
         4 -> 8          : 0        |                                        |
         8 -> 16         : 7        |**************                          |
```

Is that expeceted?

Reply:
Hmm, looks normal to me - what did you not expect? profile_blockio currently runs and updates the latency every second until _stopped_. For the GIM that currently means until deleted.

Reply:
Ops. Sorry I missed a step. I updated it. Basically, if I `stop` a gadget I assume it will stop profiling the node. But I understand I need to delete to stop profiling. I find it bit confusing. Just wondering if `flush` might be better name for `stop` but not to sure ...